ssh.go

```diff
@@ -64,13 +72,20 @@ func (c *SSHClient) parseHost(host string) error {
 		c.user = u.Username
 	}
 
-	if strings.Index(c.host, "/") != -1 {
-		return ErrConnect{c.user, c.host, "unexpected slash in the host URL"}
-	}
-
-	// Add default port, if not set
-	if strings.Index(c.host, ":") == -1 {
-		c.host += ":22"
+	c.env = c.env + `export SUP_HOST="` + c.host + `";`
+	if m, _ := url.ParseQuery(info.RawQuery); len(m) > 0 {
+		for k, vs := range m {
+			if len(vs) == 0 || vs[len(vs)-1] == "" {
+				continue
+			}
+
+			v := vs[len(vs)-1]
+			if (v[0] == '\'' && v[len(v)-1] == '\'') || (v[0] == '"' && v[len(v)-1] == '"') {
```

Comment:
I'm not sure what this complex condition is for.

Why don't you `strings.Trim()` it without these checks?

Reply:
the reason is the difference between `'` and `"` in shell
 
example:
```
ssh://root:123@456@192.168.16.10:22/~/.ssh/id_rsa?MY_VAR="aa bb cc dd"&ABC_VAR='help, run: echo $HOME'&XYZ_VAR="my home dir is $HOME"&other_var='abc$xyz'
```

if no the 'complex condition' and just use `string.Trim()`, the shell variable in c.env will be:
```
export MY_VAR="aa bb cc dd"; 
export ABC_VAR="help, run: echo $HOME"; 
export XYZ_VAR="my home dir is $HOME"; 
export other_var="abc$xyz";
```

but, I want:
```
export MY_VAR="aa bb cc dd";
export ABC_VAR='help, run: echo $HOME'; 
export XYZ_VAR="my home dir is $HOME"; 
export other_var='abc$xyz';
```

the difference:
```
mac-dev:~ kadefor$ abc='help, run: echo $HOME'
mac-dev:~ kadefor$ echo $abc
help, run: echo $HOME

mac-dev:~ kadefor$ abc="help, run: echo $HOME"
mac-dev:~ kadefor$ echo $abc
help, run: echo /Users/kadefor

mac-dev:~ kadefor$ abc='abc$xyz'
mac-dev:~ kadefor$ echo $abc
abc$xyz

mac-dev:~ kadefor$ abc="abc$xyz"
mac-dev:~ kadefor$ echo $abc
abc

```

Reply:
Sorry, I don't understand what `ssh://root:123@456@192.168.16.10:22/~/.ssh/id_rsa?MY_VAR="aa bb cc dd"&ABC_VAR='help, run: echo $HOME'&XYZ_VAR="my home dir is $HOME"&other_var='abc$xyz'` represents.

We should be parsing pure SSH connection string. Imho, there shouldn't be any path after the `[scheme://username:passwd@]host[:port]`.

Reply:
I think it is a way to set env for per host, currently. If have a new and better method in the future( save host info to struct in Supfile?), we can disable the feature. 

Need to disable it in this PR now? :-)

Reply:
you can set env vars via Supfile :)

imho we should just strip off hostUrl.Path

Reply:
currently, how to set env for per host in Supfile? (about #111)

Could I leave it as an experimental function? If not, remove it  :-D

Thank you for your review, again!

Reply:
Interesting, I don't think `sup` supports setting env vars per host right now. It supports setting env vars globally or per network.

Yes please, remove this feature for now. You can create a new issue for it and we'll see if there's any more interest for it out there. I doubt it, though.

We might think about per-host env vars, though. I like it.

Or, I think you could even set those env vars on the server side in the ~/.ssh/authorized_keys file:
command="export VAR=1; /bin/sh" ssh-key

Reply:
ok, I'll remove it. Thanks for your suggestions! :)